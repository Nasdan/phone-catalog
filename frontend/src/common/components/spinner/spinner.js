import React from 'react';
import PropTypes from 'prop-types';
import { promiseTrackerHoc } from 'react-promise-tracker';
import { ScaleLoader } from 'react-spinners';
import { spinner, loader } from './spinner.styles';

const InnerSpinner = props =>
  props.trackedPromiseInProgress && (
    <div className={spinner}>
      <div className={loader}>
        <ScaleLoader />
      </div>
    </div>
  );

  InnerSpinner.propTypes = {
  trackedPromiseInProgress: PropTypes.bool.isRequired,
};

export const Spinner = promiseTrackerHoc(InnerSpinner);
