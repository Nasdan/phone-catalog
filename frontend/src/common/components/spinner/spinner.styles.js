import { css } from 'emotion';

export const spinner = css`
  width: 100%;
  height: 100%;
`;

export const loader = css`
  display: flex;
  justify-content: center;
  align-items: center;
`;
