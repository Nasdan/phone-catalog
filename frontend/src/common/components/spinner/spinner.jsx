import React from 'react';
import PropTypes from 'prop-types';
import { promiseTrackerHoc } from 'react-promise-tracker';
import { PropagateLoader } from 'react-spinners';
import { spinner, loader } from './spinner.styles';

const InnerSpinner = props =>
  props.trackedPromiseInProgress && (
    <div className={spinner}>
      <div className={loader}>
        <PropagateLoader color="#ffffff" size={25} />
      </div>
    </div>
  );

InnerSpinner.propTypes = {
  trackedPromiseInProgress: PropTypes.bool.isRequired,
};

export const Spinner = promiseTrackerHoc(InnerSpinner);
