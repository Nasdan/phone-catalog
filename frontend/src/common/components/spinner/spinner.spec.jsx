import React from 'react';
import { shallow } from 'enzyme';
import { Spinner } from './spinner';

describe('common/components/spinner specs', () => {
  it('should render InnerSpinner component with trackedPromiseInProgress equals false', () => {
    // Arrange

    // Act
    const component = shallow(<Spinner />);

    // Assert
    expect(component).toMatchSnapshot();
  });
});
