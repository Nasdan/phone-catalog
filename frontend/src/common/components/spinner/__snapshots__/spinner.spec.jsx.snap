// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/components/spinner specs should render InnerSpinner component with trackedPromiseInProgress equals false 1`] = `
<InnerSpinner
  trackedPromiseInProgress={false}
/>
`;
